src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   Represents a text JSON value.
+    /// </summary>
+    public class JsonString : JsonNode, IEquatable<JsonString>
+    {
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonString"/> class representing the empty value.
+        /// </summary>
+        public JsonString() => Value = "";
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonString"/> class representing a specified value.
+        /// </summary>
+        public JsonString(string value) => Value = value;
+
+        /// <summary>
+        ///   Gets or sets the text value represented by the instance.
+        /// </summary>
+        public string Value { get; set; }
+
+        /// <summary>
+        ///   Converts a <see cref="string"/> to a JSON string.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonString(string value) => new JsonString(value);
+
+        /// <summary>
+        ///   Compares <paramref name="obj"/> to the value of this instance. 
+        /// </summary>
+        /// <param name="obj">The object to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the text value of this instance matches <paramref name="obj"/>,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public override bool Equals(object obj) => obj is JsonString jsonString && Value == jsonString.Value;
+
+        /// <summary>
+        ///   Calculates a hash code of this instance.
+        /// </summary>
+        /// <returns>A hash code for this instance.</returns>
+        public override int GetHashCode() => Value.GetHashCode();
+
+        /// <summary>
+        ///   Compares other JSON string to the value of this instance. 
+        /// </summary>
+        /// <param name="other">The JSON string to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the text value of this instance matches <paramref name="other"/>,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public bool Equals(JsonString other) => !(other is null) && Value == other.Value;
```

Comment:
@stephentoub Just a question: is there a difference between doing `!(other is null)` and `(other != null)`? Do we have a convention for this in corefx?

Reply:
> is there a difference between doing !(other is null) and (other != null)?

The former always does the equivalent of `!ReferenceEquals(other, null)`.  The latter will do that if `other` does not define an `operator !=`, but if `other` does, then the latter will use that custom operator.

Reply:
... unless `other` is (at compile time) `string`, in which case the compiler special cases the comparison to the null literal for `== null` and `!= null` to not invoke the operator/Equals method. (IIRC from semi-recent experimentation)